tests/integration/test_charm.py

```diff
@@ -2,11 +2,34 @@
 # See LICENSE file for licensing details.
 
 """Integration test for haproxy charm."""
+import ipaddress
+
 import pytest
+import requests
 from juju.application import Application
+from juju.client._definitions import FullStatus, UnitStatus
 
 
 @pytest.mark.abort_on_fail
 async def test_deploy(application: Application):
-    """Deploy the charm."""
-    assert application
+    """
+    arrange: Deploy the charm.
+    act: Send a GET request to the unit's ip address.
+    assert: The charm correctly response with the default page.
+    """
+    status: FullStatus = await application.model.get_status([application.name])
+    unit_status: UnitStatus = next(iter(status.applications[application.name].units.values()))
```

Comment:
I'll send you a MM thread link to more options for getting unit IP address in integration tests :)